source/grammar/qasm3.g4

```diff
@@ -455,27 +463,22 @@ externDeclaration
     : 'extern' Identifier ( LPAREN classicalTypeList? RPAREN )? returnSignature? SEMICOLON
     ;
 
-// if have extern w/ out args, is ambiguous; may get matched as identifier
-externCall
-    : Identifier LPAREN expressionList? RPAREN
+// if have function call w/ out args, is ambiguous; may get matched as identifier
+externOrSubroutineCall
+    : Identifier ( LPAREN expressionList? RPAREN )?
     ;
 
 /*** Subroutines ***/
 
 subroutineDefinition
-    : 'def' Identifier ( LPAREN classicalArgumentList? RPAREN )? quantumArgumentList?
+    : 'def' Identifier ( LPAREN anyTypeArgumentList? RPAREN )?
```

Comment:
Same comment about whether parentheses should be required in the case of no arguments...